planner/core/planbuilder.go

```diff
@@ -3019,6 +3019,8 @@ func (b *PlanBuilder) buildValuesListOfInsert(ctx context.Context, insert *ast.I
 					Value:   x.Datum,
 					RetType: &x.Type,
 				}
+			case *driver.ParamMarkerExpr:
+				expr, err = expression.ParamMarkerExpression(b.ctx, x)
```

Comment:
Where is the unnecessary memory allocating code?

Reply:
in the default case